model/user.go

```diff
@@ -11,8 +11,8 @@ import (
 type User struct {
 	CommonFields
 	Username         string     `gorm:"unique" json:"username"`
-	Email            string     `gorm:"unique" json:"email"`
-	OAuthID          string     `gorm:"unique" json:"-"`
+	Email            *string    `gorm:"unique" json:"email"`
```

Comment:
Why do you change these two fields to pointers?

Reply:
Users that get automatically created by the username header don't have an email associated with them, and only one row can have the empty string as a value because of the unique constraint. There's no such limit for SQL nulls, which are always considered to be unique, so we can have multiple automatic users.